thirdparty/github.go

```diff
@@ -2006,3 +2006,39 @@ func UpdateCheckrun(ctx context.Context, owner, repo, name string, checkRunID in
 
 	return checkRun, nil
 }
+
+func ValidateCheckRun(checkRun *github.CheckRunOutput) error {
+	if checkRun == nil {
+		return errors.New("checkRun Output is nil")
+	}
+
+	catcher := grip.NewBasicCatcher()
+
+	catcher.NewWhen(checkRun.Title == nil, "checkRun has no title")
+	summaryErrMsg := fmt.Sprintf("the checkRun '%s' has no summary", utility.FromStringPtr(checkRun.Title))
+	catcher.NewWhen(checkRun.Summary == nil, summaryErrMsg)
+
+	for _, an := range checkRun.Annotations {
+		annotationErrorMessage := fmt.Sprintf("checkRun '%s' specifies an annotation '%s' with no ", utility.FromStringPtr(checkRun.Title), utility.FromStringPtr(an.Title))
+
+		catcher.NewWhen(an.Path == nil, annotationErrorMessage+"path")
```

Comment:
Can we add a %s to `annotationErrorMessage` and use fmt.Sprintf(annotationErrorMessage, "path") for this method of constructing error messages instead?